client/state/selectors/get-media-item-server-id-from-transient-id.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Retrieves the server ID for a given transient ID and site ID.
+ *
+ * @param {object} state The current state
+ * @param {number} siteId The site ID
+ * @param {(number|string)} transientId The transient ID of the media item to get the server ID for
+ * @returns {(number|null)} The `number` server ID if it exists, null otherwise.
+ */
+export default function getMediaItemServerIdFromTransientId( state, siteId, transientId ) {
+	return state?.media?.transientItems?.[ siteId ]?.transientIdsToServerIds?.[ transientId ] ?? null;
```

Comment:
We don't technically need it, you're right. I usually prefer to have as-explict a return value as possible, so in this case it's `number | null` instead of `number | null | undefined`. Happy to remove it though as it could be unnecessary noise that would lead to confusion.

Reply:
I'm fine with keeping it :+1: